src/foam/nanos/ruler/predicate/PropertyIsClass.js

```diff
@@ -40,11 +41,11 @@ foam.CLASS({
 
       if ( getIsNew() ) {
         FObject nu  = (FObject) NEW_OBJ.f(obj);
-        return CLASS_OF(getOf().getObjClass()).f(nu.getClassInfo().getAxiomByName(getPropName()));
+         return (new IsClassOf(getOf())).f(nu.getProperty(getPropName()));
```

Comment:
getOf().getObjClass() == nu.getProperty(getPropName()).getClass(); OR
getOf().isInstance(nu.getProperty(getPropName()));

The first line is only exact match, the second handles sub-classes.

Reply:
My mistake. Carry on :(.